apis/apiextensions/v1/composition_types.go

```diff
@@ -109,7 +109,11 @@ type CompositionSpec struct {
 // +genclient
 // +genclient:nonNamespaced
 
-// A Composition specifies how a composite resource should be composed.
+// A Composition defines a template for creating multiple managed resources into
+// a single Kubernetes object.
```

Comment:
I think it's important that this captures:

* The single Kubernetes object is called an XR. i.e. That Compositions power XRs.
* The composed resources are often, but not always MRs. It's common to compose XRs, for example.

I understand that often we want to start simple and expand, but I don't want to oversimplify here in the type documentation.

On a more subjective note:

* I don't love the framing of a Composition being "a template" in this context. It's fine in the docs, but it feels reductive here. A pipeline of functions for example is arguably no longer a template.
* I'd prefer not to avoid "composes". It's in the names - Composite Resource, Composition, etc. "Creating into" feels less natural than "composing of".